core/trino-main/src/main/java/io/trino/sql/planner/planprinter/IoPlanPrinter.java

```diff
@@ -717,7 +716,9 @@ private EstimatedStatsAndCost getEstimatedStatsAndCost(TableScanNode node)
 
         private Set<ColumnConstraint> parseConstraints(TableHandle tableHandle, TupleDomain<ColumnHandle> constraint)
         {
-            checkArgument(!constraint.isNone());
+            if (constraint.isNone()) {
+                return ImmutableSet.of();
```

Comment:
That's exactly what we would return if predicate is `all`, so it's probably not the right return value for the `none` case.

Reply:
`IoPlan.TableColumnInfo` cannot currently represent a table with NONE constraint.

In general, such table should be eliminated from the query plan, this happens e.g. here
https://github.com/trinodb/trino/blob/14b901dc533eb28a6898650642b133422a1c2c5a/core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/PushPredicateIntoTableScan.java#L255-L258

We could fix the optimizer so that it happens as well in the `SELECT * FROM empty_partitioned_table` case and we probably should. However, this wouldn't eliminate the need to fix `EXPLAIN (TYPE IO)` from failing in such case, since it's generally a possible situation. That's why we deal with it also on page source level instead of failing there
https://github.com/trinodb/trino/blob/ad8663794f0e6a63cfee6b420f86a522fafc1f63/core/trino-main/src/main/java/io/trino/split/PageSourceManager.java#L63

So, back to our problem. We need to make `IoPlan.TableColumnInfo` be able to represent table scan without data, with none constraint.
I would suggest replacing `Set<ColumnConstraint> columnConstraints` field with
```
class Constraint {
  boolean isNone;
  Set<ColumnConstraint> columnConstraints;
}
```

cc @martint @sopel39 @kokosing @losipiuk @JamesRTaylor